update theme and style shared elements

mark-prins opened the pull request:
Let's just say that this Fixes #2 and Fixes #15

Made some changes to the theme, in order to support custom items and implemented some of the styling from zeplin.

joshxg commented on packages/host/src/AppDrawer.tsx:
```diff
@@ -62,21 +96,63 @@ const ListItemLink: React.FC<ListItemLinkProps> = props => {
+        component={CustomLink}
+        className={className}
+      >
+        {props.icon}
```
Maybe wrap in `<ListItemIcon>`?

joshxg commented on packages/host/src/AppDrawer.tsx:
```diff
@@ -62,21 +96,63 @@ const ListItemLink: React.FC<ListItemLinkProps> = props => {
+  );
 
   return (
     <li>
```
this `li` seems unneeded?

mark-prins replied:
I took them out and bathed in the delight of minimalism, and then noticed that it was all wrapped in a `List` so added them back in to be syntactically complete 🤷‍♂️

joshxg replied:
Ah I just read the docs - how silly that `*L*ist*I*tem` isn't rendering a `<*l**i*>` 😢

joshxg reviewed (approved):
It's amazing!  Looking great

Just a small thing with the text when the drawer is closed:
![image](https://user-images.githubusercontent.com/35858975/127824483-ab33c33f-78d6-4522-b24b-0fc8cb340764.png)

Wrapping the icon in `ListItemIcon` seems to fix it, but then the selected and hovering background gets a bit ugly.. might need to use the drawer open state to conditionally render the text?

Spent some time figuring out theme stuff and made some useless comments about putting things in the theme, which could be done later, if they should be put there..?

Also maybe need to ask Manu to put an 'open' button? 🤔

mark-prins commented:
> Also maybe need to ask Manu to put an 'open' button?

His thought with the right hand drawer was to make it responsive; if the screen size was big enough it would be expanded otherwise collapsed.. though it did have the option to open & close. 
I thought to make the msupply guy open & close this one. at least for now.

mark-prins commented:
> Just a small thing with the text when the drawer is closed

yeah, I did sort of give up on that; will do it now.
I was avoiding the `ListItemIcon` in order to remove some of the overhead. and that was doing stuff which I didn't want.. Will have a look.

mark-prins commented on packages/host/src/AppDrawer.tsx:
```diff
@@ -1,38 +1,69 @@
+  },
+  drawerMenuItemSelected: {
+    backgroundColor: '#fff!important',
+    borderRadius: 16,
```
that gives a radius of `4` - I went with the specific value here, thinking that this lozenge menu item will only appear in the drawer. could well be wrong about that.

mark-prins commented:
that was fun.. 
- I've added in the lower menu items
- and the divider between them
- hidden the text when the drawer is closed
- updated the style of the items when closed
- added back the `ListItemIcon`
- updated the mSupply Guy, wrapping him in an `IconButton` and toggling the drawer
- Used material shadows ( which aren't quite right, but I like the idea )
- Added colours to the palette
- Added mixins

joshxg commented:
> that was fun..

Sarcasm or serious? 🤔
